tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,197 @@
+        let path = kit_path.join(&kit_filename);
+
+        let out = docker!(["load", format!("--input={}", path.display()).as_str(),]);
+        let out = String::from_utf8_lossy(&out);
+        let digest_expression =
+            Regex::new("(?<digest>sha256:[0-9a-f]{64})").context(error::RegexSnafu)?;
+        let caps = digest_expression.captures(&out).unwrap();
+        let digest = &caps["digest"];
+
+        let arch_specific_target_uri = format!(
+            "{}/{}:{}-{}",
+            vendor_registry_uri, kit_name, &kit_version, arch
+        );
+
+        docker!(["tag", digest, &arch_specific_target_uri,]);
+
+        info!(
+            "Pushing kit image for platform {} to {}",
+            arch, &arch_specific_target_uri
+        );
+        docker!(["push", &arch_specific_target_uri,]);
+
+        let docker_arch =
+            DockerArchitecture::try_from(arch).context(error::InvalidArchitectureSnafu)?;
+        platform_images.push((docker_arch, arch_specific_target_uri.clone()));
+    }
+
+    let target_uri = format!("{}/{}:{}", vendor_registry_uri, kit_name, kit_version);
+
+    let images: Vec<&str> = platform_images
+        .iter()
+        .map(|(_, image)| image.as_str())
+        .collect();
+
+    let mut manifest_create_args = vec!["manifest", "create"];
+    if vendor.insecure {
+        manifest_create_args.push("--insecure")
+    }
+    manifest_create_args.push(&target_uri);
+    manifest_create_args.extend_from_slice(&images);
+    docker!(manifest_create_args);
+
+    for (arch, image) in platform_images.iter() {
+        docker!([
+            "manifest",
+            "annotate",
+            format!("--arch={}", arch).as_str(),
+            &target_uri,
+            image,
+        ]);
+    }
+
+    info!("Pushing kit to {}", &target_uri);
+    docker!(["manifest", "push", &target_uri,]);
+
+    info!("Successfully published kit to {}", target_uri);
+
+    // Cleans up the transient images. These don't serve a purpose outside of publishing the kit
+    // and take up space.
+    info!("Cleaning up local images");
+    docker!(["manifest", "rm", &target_uri]);
+    for (_, image) in platform_images.iter() {
+        docker!(["rmi", image]);
+    }
+
+    Ok(())
+}
+
+fn find_version(kit_path: &Path, kit_name: &str) -> Result<String> {
+    let pattern = format!(
+        "{}-(?<version>v[0-9.]+-[0-9a-f]+(-dirty)?)-(aarch64|x86_64).tar",
+        kit_name
+    );
+    let expression = Regex::new(&pattern).context(error::RegexSnafu)?;
+
+    let file = read_dir(kit_path)
+        .context(error::IoSnafu)?
+        .map(|entry| {
+            entry
+                .unwrap()
+                .path()
+                .file_name()
+                .unwrap()
+                .to_string_lossy()
+                .to_string()
+        })
+        .collect::<Vec<_>>()
+        .into_iter()
+        .find(|file_name| expression.is_match(file_name))
+        .context(error::NoArchiveSnafu { path: kit_path })?;
```

Comment:
The way we handle this elsewhere in `pubsys` is to take a command line argument that exactly specifies the version - that way we're 100% sure we're pushing the artifact that matches the current repo state.

So we should arrange for `twoliter` to pass in the full path or enough data to reconstruct the file name:
```
${KIT:?}-v${VERSION_ID:?}-${BUILD_ID:?}-${ARCH:?}"
```